File: frontend/src/types/report.ts
export type TestStatus = 'within_provided_range' | 'outside_provided_range' | 'unable_to_determine';

export interface ExplanationItem {
  test_explanation: string;
  simple_meaning: string;
  attention_reason: string;
  professional_discussion: string;
}

export interface TestResultItem {
  name: string;
  value: number | string;
  numeric_value?: number | null;
  unit: string;
  reference_low?: number | null;
  reference_high?: number | null;
  reference_text: string;
  status: TestStatus;
  explanation?: ExplanationItem;
}

export interface ReportSummary {
  within_range: number;
  outside_range: number;
  unable_to_determine: number;
  total_tests: number;
  patient_summary_text: string;
}

export interface ReportResponse {
  report_id: string;
  report_type: string;
  report_date: string;
  patient_name?: string;
  language: string;
  summary: ReportSummary;
  tests: TestResultItem[];
  is_demo: boolean;
  status: string;
  created_at: string;
}

export interface DoctorSummaryItem {
  test_name: string;
  reported_value: number | string;
  unit: string;
  reference_range: string;
  discussion_point: string;
}

export interface DoctorSummaryResponse {
  report_id: string;
  report_type: string;
  report_date: string;
  outside_range_count: number;
  items: DoctorSummaryItem[];
  clinical_note: string;
  disclaimer: string;
}

export type SupportedLanguage = 'en' | 'ta' | 'hi' | 'te' | 'ml' | 'kn';

export interface LanguageOption {
  code: SupportedLanguage;
  label: string;
  nativeLabel: string;
  isFullySupported: boolean;
}

export const SUPPORTED_LANGUAGES: LanguageOption[] = [
  { code: 'en', label: 'English', nativeLabel: 'English', isFullySupported: true },
  { code: 'ta', label: 'Tamil', nativeLabel: 'தமிழ்', isFullySupported: true },
  { code: 'hi', label: 'Hindi', nativeLabel: 'हिन्दी', isFullySupported: false },
  { code: 'te', label: 'Telugu', nativeLabel: 'తెలుగు', isFullySupported: false },
  { code: 'ml', label: 'Malayalam', nativeLabel: 'മലയാളം', isFullySupported: false },
  { code: 'kn', label: 'Kannada', nativeLabel: 'ಕನ್ನಡ', isFullySupported: false },
];

File: backend/app/schemas/report.py
from enum import Enum
from typing import Optional, List, Union
from pydantic import BaseModel, Field

class TestStatus(str, Enum):
    WITHIN_PROVIDED_RANGE = "within_provided_range"
    OUTSIDE_PROVIDED_RANGE = "outside_provided_range"
    UNABLE_TO_DETERMINE = "unable_to_determine"

class ExplanationItem(BaseModel):
    test_explanation: str = Field(..., description="What this test generally measures")
    simple_meaning: str = Field(..., description="Plain-language restatement of the status")
    attention_reason: str = Field(default="", description="Why this is/isn't highlighted (empty if within range)")
    professional_discussion: str = Field(default="", description="A neutral prompt to discuss with a clinician, or empty string")

class ParsedTest(BaseModel):
    name: str
    value: Union[float, int, str]
    numeric_value: Optional[float] = None
    unit: str = ""
    reference_low: Optional[float] = None
    reference_high: Optional[float] = None
    reference_text: str = ""
    status: TestStatus = TestStatus.UNABLE_TO_DETERMINE

class TestResultItem(BaseModel):
    name: str
    value: Union[float, int, str]
    numeric_value: Optional[float] = None
    unit: str = ""
    reference_low: Optional[float] = None
    reference_high: Optional[float] = None
    reference_text: str = ""
    status: TestStatus
    explanation: Optional[ExplanationItem] = None

class ReportSummary(BaseModel):
    within_range: int = 0
    outside_range: int = 0
    unable_to_determine: int = 0
    total_tests: int = 0
    patient_summary_text: str = ""

class ReportResponse(BaseModel):
    report_id: str
    report_type: str = "General Lab Report"
    report_date: str = ""
    patient_name: Optional[str] = "Patient"
    language: str = "en"
    summary: ReportSummary
    tests: List[TestResultItem]
    is_demo: bool = False
    status: str = "ready"
    created_at: str = ""

class TranslateRequest(BaseModel):
    target_language: str = Field(..., description="Language code e.g. 'ta', 'en', 'hi', 'te', 'ml', 'kn'")

class DoctorSummaryItem(BaseModel):
    test_name: str
    reported_value: Union[float, int, str]
    unit: str
    reference_range: str
    discussion_point: str

class DoctorSummaryResponse(BaseModel):
    report_id: str
    report_type: str
    report_date: str
    outside_range_count: int
    items: List[DoctorSummaryItem]
    clinical_note: str
    disclaimer: str
